object-handlers.go

```diff
@@ -144,7 +144,10 @@ func (api objectAPIHandlers) GetObjectHandler(w http.ResponseWriter, r *http.Req
 	// Reads the object at startOffset and writes to mw.
 	if err := api.ObjectAPI.GetObject(bucket, object, startOffset, length, w); err != nil {
 		errorIf(err, "Unable to write to client.")
-		// Do not send error response here, client would have already died.
+		// In case headers haven't been written to client, unset the headers and send an error response.
+		unsetObjectHeaders(w)
+		apiErr := toAPIErrorCode(err)
+		writeErrorResponse(w, r, apiErr, r.URL.Path)
```

Comment:
You cannot do this, response writer cannot be written again once you have started writing.